src/plugins/locale/index.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable dot-notation */
+import type { EsDayPlugin, Locale, UnitType } from 'esday'
+import * as C from '~/core/constant'
+import { prettyUnit, undefinedOr } from '~/core/utils'
+import en from '~/locales/en'
+
+const LocaleStore: Map<string, Locale> = new Map()
+
+let $localeGlobal = 'en'
+
+export function getLocale(name: string): Locale {
+  return LocaleStore.get(name) || en
+}
+
+export function registerLocale(locale: Locale, rename?: string): void {
+  LocaleStore.set(rename || locale.name, locale)
+}
+
+declare module 'esday' {
+/*   interface EsDay {
+    $locale: () => Locale
+  } */
+
+  interface EsDay {
+    locale: (localeName: string) => EsDay
+  }
+
+  interface EsDayFactory {
+    /**
+     * use locale as global
+     */
+    locale: (localeName: string) => EsDayFactory
+    /**
+     * register locale
+     */
+    registerLocale: (locale: Locale, newName?: string) => EsDayFactory
+  }
+}
+
+function getSetPrivateLocaleName(inst: any, v?: string): string {
```

Comment:
+ can we use EsDay as the type instead of any?
+ you know I like 'speaking' variable names :-) `v` says nothing to me and even the long form 'value' doesn't really help (even when it is an internal function).

Reply:
Ok. I will update it.